File: wk04/verification/day_to_year.py
from calendar import isleap

ORIGIN_YEAR = 1970

def day_to_year(days):
    year = ORIGIN_YEAR

    while days > 365:
        if isleap(year):
            if days > 366:
                days -= 366
                year += 1
        else:
            days -= 365
            year += 1

    return year

File: wk08/b.coverage/day-to-year.ts
const ORIGIN_YEAR = 1970;

const isLeap = (y: number) => new Date(y, 1, 29).getDate() === 29;

export const dayToYear = (days) => {
  let year = ORIGIN_YEAR;
  while (days > 365) {
    if (isLeap(year)) {
      if (days > 366) {
        days -= 366;
        year += 1;
      } else {
        // Catches Bug
        break;
      }
    } else {
      days -= 365;
      year += 1;
    }
  }
  return year;
};
